Tests/Acceptance/CheckoutTestCase.php

```diff
@@ -0,0 +1,146 @@
+<?php
+/**
+ * Shop System Plugins:
+ * - Terms of Use can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/_TERMS_OF_USE
+ * - License can be found under:
+ * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
+ */
+
+namespace Wirecard\Oxid\Tests\Acceptance;
+
+use Wirecard\Oxid\Model\Transaction;
+
+/**
+ * Acceptance test class for testing checkout flows.
+ */
+abstract class CheckoutTestCase extends BaseAcceptanceTestCase
+{
+    /**
+     * @var Wirecard\Oxid\Model\PaymentMethod
+     */
+    protected $paymentMethod;
+
+    /**
+     * @inheritdoc
+     */
+    protected function setUp()
+    {
+        parent::setUp();
+
+        $this->paymentMethod = $this->getPaymentMethod();
+
+        $this->insertMockData();
+        $this->activatePaymentMethod();
+    }
+
+    /**
+     * Payment method getter.
+     */
+    abstract public function getPaymentMethod();
+
+    /**
+     * Activates the payment method.
+     */
+    public function activatePaymentMethod()
+    {
+        $this->executeSql("UPDATE `oxpayments` SET `OXACTIVE` = '1'
+            WHERE `OXID` = '{$this->paymentMethod::getName(true)}'");
+    }
+
+    /**
+     * Sets a payment method's payment action to "purchase".
+     */
+    public function setPaymentActionPurchase()
+    {
+        $this->executeSql("UPDATE `oxpayments` SET `WDOXIDEE_TRANSACTIONACTION` = '" . Transaction::ACTION_PAY .
+            "' WHERE `OXID` = '{$this->paymentMethod::getName(true)}'");
+    }
+
+    /**
+     * Sets a payment method's payment action to "authorize".
+     */
+    public function setPaymentActionAuthorize()
+    {
+        $this->executeSql("UPDATE `oxpayments` SET `WDOXIDEE_TRANSACTIONACTION` = '" . Transaction::ACTION_RESERVE .
+            "' WHERE `OXID` = '{$this->paymentMethod::getName(true)}'");
+    }
+
+    /**
+     * Inserts mock data to the database.
+     */
+    public function insertMockData()
+    {
+        foreach ($this->getMockData() as $table => $entries) {
+            foreach ($entries as $fields) {
+                $columns = '`' . implode('`, `', array_keys($fields)) . '`';
+                $values = '\'' . implode('\', \'', array_values($fields)) . '\'';
+
+                $this->executeSql("INSERT INTO `{$table}` ({$columns}) VALUES ({$values})");
+            }
+        }
+    }
+
+    /**
+     * Logs a mock user into the frontend.
+     */
+    public function loginMockUserToFrontend()
+    {
+        $this->loginInFrontend(
+            $this->getMockData('oxuser.0.OXUSERNAME'),
+            $this->getMockData('oxuser.0.OXUSERNAME')
```

Comment:
The password is also the username?

Reply:
Is it better to also add a plain password?

Reply:
It is the same because we set the values from the `mock-data.json` file directly to the database – for the sake of simplicity. If we would add a plain text password field, the database insertion would fail.

I refactored the code a little and added a comment so that this is more obvious.